pyomo/contrib/mcpp/pyomo_mcpp.py

```diff
@@ -0,0 +1,361 @@
+# Note: the self.mcpp.* functions are all C-style functions implemented
+# in the compiled MC++ wrapper library
+# Note: argument to pow must be an integer
+from __future__ import division
+
+import ctypes
+import logging
+import os
+
+from pyomo.core import value, Expression
+from pyomo.core.base.block import SubclassOf
+from pyomo.core.expr.current import identify_variables
+from pyomo.core.expr.expr_pyomo5 import (
+    AbsExpression, LinearExpression, NegationExpression, NPV_AbsExpression,
+    NPV_ExternalFunctionExpression, NPV_NegationExpression, NPV_PowExpression,
+    NPV_ProductExpression, NPV_ReciprocalExpression, NPV_SumExpression,
+    NPV_UnaryFunctionExpression, PowExpression, ProductExpression,
+    ReciprocalExpression, StreamBasedExpressionVisitor, SumExpression,
+    UnaryFunctionExpression, nonpyomo_leaf_types
+)
+from pyomo.core.kernel.component_map import ComponentMap
+
+logger = logging.getLogger('pyomo.contrib.mcpp')
+
+path = os.path.dirname(__file__)
+
+
+def mcpp_available():
+    """True if the MC++ shared object file exists. False otherwise."""
+    return os.path.isfile(path + '/mcppInterface.so')
```

Comment:
This is fine for development, but not for end-users, who may not have write access to this directory in order to insert the mcppInterface.so file.  Eventually we will want to make this work for Windows / OSX platforms, too.  I am starting to collect some useful utilities for managing compiled extensions in the [common-downloader](https://github.com/jsiirola/pyomo/tree/common-downloader) branch.  If you have use-cases, please pass them on.